crates/nu-command/src/platform/input_keys.rs

```diff
@@ -0,0 +1,136 @@
+use core::time::Duration;
+use crossterm::{event::poll, event::Event, event::KeyCode, event::KeyEvent, terminal};
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, IntoPipelineData, PipelineData, ShellError, Signature, Span, Value,
+};
+use std::io::{stdout, Write};
+
+#[derive(Clone)]
+pub struct InputKeys;
+
+impl Command for InputKeys {
+    fn name(&self) -> &str {
+        "input-keys"
+    }
+
+    fn usage(&self) -> &str {
+        "Get input from the user."
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("input-keys").category(Category::Platform)
+    }
+
+    fn run(
+        &self,
+        _engine_state: &EngineState,
+        _stack: &mut Stack,
+        _call: &Call,
+        _input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        // println!("Ready to print events (Abort with ESC):");
+        println!("You have 5 seconds to hit a key combination:");
+        Ok(print_events()?.into_pipeline_data())
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Get one key event",
+            example: "input-keys",
+            result: None,
+        }]
+    }
+}
+
+pub fn print_events() -> Result<Value, ShellError> {
+    stdout().flush()?;
+    terminal::enable_raw_mode()?;
+    let result = print_events_helper();
```

Comment:
What happens if you put the loop around this?

Reply:
(line 50 specifically)

Reply:
I guess I'm not sure how to do that. I did this and it doesn't print anything when I hit a key combination.
```rust
pub fn print_events() -> Result<Value, ShellError> {
    stdout().flush()?;
    terminal::enable_raw_mode()?;
    loop {
        let event = crossterm::event::read()?;
        if event == Event::Key(KeyCode::Esc.into()) {
            break;
        }

        print_events_helper(event)?;
    }
    terminal::disable_raw_mode()?;

    Ok(Value::nothing(Span::test_data()))
}
```

Reply:
took a different route.